internal/exec/stages/files/filesystemEntries.go

```diff
@@ -17,15 +17,64 @@ package files
 import (
 	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
+	"regexp"
 	"sort"
 	"strings"
 
 	"github.com/coreos/ignition/v2/config/v3_1_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 	"github.com/coreos/ignition/v2/internal/log"
+
+	"github.com/vincent-petithory/dataurl"
 )
 
+// createCrypttabEntries creates entries inside of /etc/crypttab for LUKS volumes.
```

Comment:
I think `/etc/crypttab` is a bit like `/etc/fstab` in being kind of a "legacy" format.  systemd contains generators that make units from them.

I'm OK with this as is; there are some advantages to it too, but just noting.